addons/sale/models/sale.py

```diff
@@ -492,7 +492,7 @@ def _prepare_invoice(self):
         a clean extension chain).
         """
         self.ensure_one()
-        journal = self.env['account.move'].with_context(force_company=self.company_id.id, default_type='out_invoice')._get_default_journal()
+        journal = self.env['account.move'].with_context(default_company_id=self.company_id.id, default_type='out_invoice')._get_default_journal()
```

Comment:
This part could be completly removed since an error will be raised accounting-side if the journal is missing

Reply:
This cannot be removed, since the method `_get_default_journal` use this context key to add a company domain on the search for a journal. In a multi-company settings, this method will return a journal even if it's not in the correct company. The "useless diff" a few lines later ensure that the journal we create the invoice in (and therefore, the company) is the same as the order.